pptxApp.py

```diff
@@ -0,0 +1,158 @@
+    :return: clean version of the response (string)
+    """
+    cleaned_text = re.sub(r"\n", "", text)
+    cleaned_text = cleaned_text.encode("ascii", "ignore").decode("utf-8")
+    return cleaned_text.strip()
+
+
+# --------------------------------------------------------------------------------------------------------------------
+def save_explanations(explanations, presentation_path):
+    """
+    this method receives a list of strings each string implements an explanation, creates a JSON file and appends
+    the explanations to the file, in the format of (slide#: explanations).
+    :param: explanations: list of explanation retrieved from the API (List of strings)
+    :return:
+    """
+    presentation_name = os.path.basename(presentation_path)  # get the base name of power-point from path
+    presentation_name = os.path.splitext(presentation_name)[0]  # remove file's extension
+    output_file = f"{presentation_name}_explanations.json"  # name the output file, with the name of power-point
+    slide_explanations = {}
+
+    for slide_num, explanation in enumerate(explanations, start=1):
+        slide_key = f"slide{slide_num}"
+        slide_explanations[slide_key] = explanation
+
+    try:
+        if not os.path.exists(output_file):  # create file if it does not exist.
+            open(output_file, 'w').close()
+
+        with open(output_file, 'w') as file:
+            json.dump(slide_explanations, file, indent=4)
+        print(f"Explanations saved to {output_file}")
+    except IOError as error:
+        print(f"{ERROR_MESSAGE} Error saving explanations: {str(error)}")
+
+
+# --------------------------------------------------------------------------------------------------------------------
+async def main(presentation_path):
+    """
+    this is the main method, uses asyncio handling to await the requests sent to the openai API, and handles saving
+    the responses to the output file.
+    :param: presentation_path: string of a power-point path. (string)
+    :return:
+    """
+    explanations = await parse_presentation(presentation_path)
+    save_explanations(explanations, presentation_path)
+
+
+# --------------------------------------------------------------------------------------------------------------------
+if __name__ == "__main__":
+    """
+    the controller of the app, handles the inputs from the user, prints relevant messages and runs the main program
+    """
+    presentation_path = input("Please provide the path of the PowerPoint presentation: ")
+    print("Processing PowerPoint...")  # inform the user to wait until the program finishes running
+    start_time = time.time()  # compute time of execution
+
+    asyncio.run(main(presentation_path))  # run main program
+
+    end_time = time.time()
+    execution_time = end_time - start_time
+    minutes, seconds = divmod(execution_time, 60)
+    print(f"Execution time: {minutes:.0f} minutes {seconds:.2f} seconds")  # print total time of execution
```

Comment:
it's great to check the time taken.
what about catching RunTime errors along the code?